### DISCUSSION TITLE
Boot time for E51 and U54

### DISCUSSION THREAD

**Discussion by MarkAtMicrochip** [polarfire-soc/polarfire-soc-documentation#397] [Bare metal embedded software]
What should the expected boot time for a bare metal application? I have a customer who is seeing ~5s. Is it possible to get that under 1s? Thanks!

**Comment by rockoguido** [comment]
Hello

Yes using mpfs_hal_ddr_demo have had the boot time for this demo with DDR training execute in under 200ms
The bottleneck is typically DDR training and initialization
Can share details if relevant


Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
From: MarkAtMicrochip ***@***.***>
Sent: Thursday, January 25, 2024 1:21:59 PM
To: polarfire-soc/polarfire-soc-documentation ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [PolarFire-SoC] Boot time for E51 and U54 (Discussion #397)

EXTERNAL EMAIL: Do not click links or open attachments unless you know the content is safe

What should the expected boot time for a bare metal application? I have a customer who is seeing ~5s. Is it possible to get that under 1s? Thanks!

**Comment by MarkAtMicrochip** [comment]
Yes, details are important. The customer I'm working with is booting from eNVM.

**Comment by rockoguido** [comment]
ok, the boot sequence.   does it start with HSS booting the image from eNVM??   
We were targeting mpfs_hal_ddr_demo which of course has a bootloader in it also, here was the sequence showing numbers before and after....

Baseline at 9400ms

enabled LIBERO_FAST_START - now at 3500ms
            see mss_sw_config.h   (under platform_config for the board being used)
            
removed DEBUG_DDR_INIT now at 2500ms
            also in the same file referenced above
          for HSS you will find this in the same file as above..

enabled SKIP_VERIFY_PATTERN_IN_CACHE now at 147ms
            see mss_ddr.c, user needs to enable this, same file in HSS 

So the net is the work we did for mpfs_hal_ddr_demo applies also to HSS, which makes sense since they share a lot of code.  

If you have any questions feel free to ask

thx
Dave
